### DISCUSSION TITLE
CDV (CompileDataView) current version?

### DISCUSSION THREAD

**Discussion by cmidgley** [Moddable-OpenSource/moddable#1361] [General]
CDV is located in two places - in `Moddable-OpenSource/moddable` within `tools` and also `phoddie/compileDataView`.  I took a quick look and it seems the CLI is better in the `moddable` repo, but the `compileDataView` code is more up-to-date in the `phoddie` version (and that version has features that I depend on).  I did not do a comparison / trial merge so there may be more differences.

Perhaps a more careful look at differences and merge into `moddable`?  (and maybe archive the `phoddie/compileDataView` repo?)  Just a suggestion, as I use the `phoddie` version right now but discovered this when a change to my build system starting running the `cdv` from moddable instead and broke things.

**Comment by phoddie** [comment]
My intention has been that CDV have its own repository, currently `phoddie/compileDataView`. The Moddable SDK takes a snapshot as needed. There shouldn't' be changes made to CDV in the Moddable SDK apart from that. It may well be that the Moddable SDK snapshot is out of date. It has been a while.

The biggest open project on CDV is to work through and merge your changes to support a variable length array at the end of the structure. That's not trivial (or it would be done already). I think it is realistic that it could be done in July.
